src/ucs/sys/string.c

```diff
@@ -133,10 +133,12 @@ const char* ucs_sockaddr_str(const struct sockaddr *sock_addr, char *str, size_t
     case AF_INET:
         addr_in = (struct sockaddr_in *) sock_addr;
         inet_ntop(AF_INET, &addr_in->sin_addr, str, max_size);
+        snprintf(str + strlen(str), max_size, ":%d", ntohs(addr_in->sin_port));
```

Comment:
not sure I got the whole algorithm...
but shouldn't max_size be reduced by strlen(str) prior to call snprintf?